### ISSUE TITLE
False positive for no-top-level-browser-globals with compound expression

### ISSUE THREAD

**Issue by gurpreetatwal** [sveltejs/eslint-plugin-svelte#1298]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.33.0

### What version of `eslint-plugin-svelte` are you using?

3.11.0

### What did you do?

<details>
<summary>Configuration</summary>

```
import js from "@eslint/js";
import sveltePlugin from "eslint-plugin-svelte";
import svelteParser from "svelte-eslint-parser";
import tseslint from "typescript-eslint";

export default tseslint.config(
  js.configs.recommended,
  ...tseslint.configs.recommended,
  ...sveltePlugin.configs["flat/recommended"],
  {
    files: ["**/*.svelte", "*.svelte"],
    languageOptions: {
      globals: {
        window: false,
        console: false,
      },
      parser: svelteParser,
      parserOptions: {
        parser: {
          ts: "@typescript-eslint/parser",
        },
        extraFileExtensions: [".svelte"],
      },
    },
    rules: {
     'svelte/no-top-level-browser-globals': 'error',
    }
  }
);
```
</details>

```svelte
<script>
  import { browser, dev } from '$app/environment';

  const notification = browser && dev
    ? 'Notification' in window 
    : false

  if (notification) console.log('hi');
</script>

{#if browser && dev}
  <div>
    <pre>Notification: {'Notification' in window}</pre>
  </div>
{/if}
```

### What did you expect to happen?

No errors to be reported since the access is guarded by {#if browser}

### What actually happened?

```
✕Unexpected top-level browser global variable "window".
    [(svelte/no-top-level-browser-globals)](https://sveltejs.github.io/eslint-plugin-svelte/rules/no-top-level-browser-globals/)
    [5:25-31]
✕Unexpected top-level browser global variable "window".
    [(svelte/no-top-level-browser-globals)](https://sveltejs.github.io/eslint-plugin-svelte/rules/no-top-level-browser-globals/)
    [13:43-49]
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNqNU8mSmzAQ/ZUuJTXYLgN3xpmkKsstyz3kwEBD5MgSJcnLFMW/p4WEYWwnMxwAqd/r5XV3x4wu069PH9WuVRKlTcwBhUWWsY0pNW/tQy4BOFm1hQ4etToa1Guo8AA91FrtIHpbtG2K8sC1kjvyEd3n0rFKJY0FqSyveVlYriS8Gz04O8B7iL7NzBFwCUcuK3UED8igLoRB74/XsJh7Ww4RlMBEqGYR/ebRkiJv0nPiuezeECmEhLs7l3bvXG0qfhgqo99W48M8iwy6f2XVb1KHHjyk3kWX8rpna4ZGcJKPMqp5k2wNKRhU2xqvU84+eFC6NTmjTIPdK/5D7BuKE5AeGLfDZewRNziFdnUFjr+LR+pgm3Os8aYRb59a9FIFzgDOJZ4GeIV1sRcTLZS2cMVvTTiZRGOpdtT1Cqu1MyVJcsG4iZkXPeJ+5qwWhU1n8Jz9Ghid71XNBZoMCLhapavgJGdrqmY6eQaAKGSzLxr83rouEi04AWiEeqSxmt1AaHCYt+DBPWHELg39+c8LnT1ryYXxOoOJNrsC0jpzY3LVmXTs5iyxKQMAPFldfCFxPp8sSuODkUpXmky08av3g6RjGpFnpFLFVrWxQDrGYYHioFuUQYRaKx2NvtyHXst7WoS2KP+Q6rQCStISDI79sOXMVbdTFYUMpeSMNvITtq7ZsuRIi3HO5fm+OCoNBxoa0xB2XJP/Gi936Cb0FcbLvboJnRr3IuC8c3Nc0LFn/V8w7N97

### Additional comments

Related to #1251 and #1252
